kobo/apps/audit_log/tests/test_signals.py

```diff
@@ -137,3 +130,65 @@ def test_loginas(self):
+    """
+    Test signals around the creation and assignment of submission groups
+    """
+
+    fixtures = ['test_data']
+
+    def test_new_group_for_new_submission(self):
+        self.assertEqual(AuditLog.objects.count(), 0)
+        user = User.objects.get(username='someuser')
+        new_submission = SubmissionAccessLog.objects.create(user=user)
+        # make sure a new group was made and assigned as the submission group for the submission
+        self.assertEqual(SubmissionGroup.objects.count(), 1)
+        group_log = SubmissionGroup.objects.first()
+        self.assertEqual(group_log.user.id, user.id)
+        self.assertEqual(new_submission.submission_group.id, group_log.id)
+        self.assertEqual(group_log.date_created, new_submission.date_created)
+
+    # make sure the time limit is long enough that the logs *could* be grouped together
+    # if the users matched
+    @override_settings(ACCESS_LOG_SUBMISSION_GROUP_TIME_LIMIT_MINUTES=60)
+    def test_new_group_for_new_submission_if_existing_groups_have_wrong_user(
+        self,
+    ):
+        user1 = User.objects.get(username='someuser')
+        user2 = User.objects.get(username='anotheruser')
+        # create a new submission with the first user
+        user1_submission = SubmissionAccessLog.objects.create(user=user1)
+        # make sure that created a new group
+        user1_group_query = SubmissionGroup.objects.filter(user=user1)
+        self.assertEqual(user1_group_query.count(), 1)
+        user1_group = user1_group_query.first()
+        self.assertEqual(user1_submission.submission_group.id, user1_group.id)
+
+        # create a submission with the second user
+        user2_submission = SubmissionAccessLog.objects.create(user=user2)
+        user2_group_query = SubmissionGroup.objects.filter(user=user2)
+        self.assertEqual(user2_group_query.count(), 1)
+        new_group = user2_group_query.first()
+        self.assertEqual(user2_submission.submission_group.id, new_group.id)
+
+    @override_settings(ACCESS_LOG_SUBMISSION_GROUP_TIME_LIMIT_MINUTES=5)
+    def test_new_group_for_new_submission_if_existing_groups_too_old(self):
+        user = User.objects.get(username='someuser')
+        ten_minutes_ago = timezone.now() - timedelta(minutes=10)
+        old_submission = SubmissionAccessLog.objects.create(
+            user=user, date_created=ten_minutes_ago
+        )
+
+        # make sure that created a new group
+        group_query = SubmissionGroup.objects.filter(user=user)
+        self.assertEqual(group_query.count(), 1)
+        old_group = group_query.first()
+        self.assertEqual(old_submission.submission_group.id, old_group.id)
+
+        # new submission
+        new_submission = SubmissionAccessLog.objects.create(user=user)
+        self.assertEqual(group_query.count(), 2)
+        new_group = group_query.order_by('-date_created').first()
+        self.assertEqual(new_submission.submission_group.id, new_group.id)
```

Comment:
even if `group_query.count() == 2`, I think it could be great to validate ` new_group.id != old_group.id` just in case.